h2integrate/postprocess/mapping.py

```diff
@@ -0,0 +1,852 @@
+
+    offset_text = cbar.ax.xaxis.get_offset_text()
+    offset_text.set_fontsize(map_preferences.colorbar_tick_label_font_size)
+
+    # Dynamically set the exponential notation text x position based on the colorbar's width
+    # Defaults to +2.5% relative to the colorbar width, normalized to the inset axis of the colorbar
+    # Ex: colorbar_width = 20%, dynamic offset = (22.5/20) = 1.125
+    # NOTE: hardcoding this for ease of handling and reducing inputs, can be changed if needed
+    colorbar_width = float(map_preferences.colorbar_width[:-1])
+    dyn_exp_notation_x_offset = (colorbar_width + 2.5) / colorbar_width
+    offset_text.set_position((dyn_exp_notation_x_offset, np.nan))
+    # Set the expontential notation y position centered on the colorbar
+    # NOTE: hardcoding this for ease of handling and reducing inputs, can be changed if needed
+    cbar.ax.xaxis.OFFSETTEXTPAD = -24
+
+    # Calculate appropriate bounds for map based on coordinates of data used in plots
+    coord_range_dict = calculate_geodataframe_total_bounds(*gdfs_for_bounds)
+
+    # Set basemap extent based on calculated bounds and padding fractions
+    left_pad = coord_range_dict["x_range"] * map_preferences.basemap_leftpad
+    right_pad = coord_range_dict["x_range"] * map_preferences.basemap_rightpad
+    upper_pad = coord_range_dict["y_range"] * map_preferences.basemap_upperpad
+    lower_pad = coord_range_dict["y_range"] * map_preferences.basemap_lowerpad
+
+    ax.set_xlim(coord_range_dict["min_x"] - left_pad, coord_range_dict["max_x"] + right_pad)
+    ax.set_ylim(coord_range_dict["min_y"] - lower_pad, coord_range_dict["max_y"] + upper_pad)
+    ax.set_axis_off()
+    ax.set_title(map_preferences.figure_title)
+
+    # Plot basemap with contextily
+    ctx.add_basemap(
+        ax,
+        crs=map_preferences.web_map_crs,
+        source=map_preferences.basemap_provider,
+        zoom=map_preferences.basemap_zoom,
+    )
+
+    # Show the plot if show_plot == True
+    # NOTE: when plotting multiple layers, set this to True only when plotting the last layer
+    if show_plot:
+        plt.show()
+
+    # Save figure if save_plot_fpath is present
+    if save_plot_fpath is not None:
+        if isinstance(save_plot_fpath, str):
+            save_plot_fpath = Path(save_plot_fpath)
+        fig.savefig(save_plot_fpath)
+
+    return fig, ax, results_gdf
+
+
+def plot_straight_line_shipping_routes(
+    shipping_coords_fpath: Path | str,
+    shipping_route: list[str],
+    latitude_var_name: str | None = None,
+    longitude_var_name: str | None = None,
+    *,
+    fig: plt.Figure | None = None,
+    ax: plt.Axes | None = None,
+    base_layer_gdf: gpd.GeoDataFrame
+    | list[gpd.GeoDataFrame]
+    | tuple[gpd.GeoDataFrame, ...]
+    | None = None,
+    show_plot: bool = False,
+    save_plot_fpath: Path | str | None = None,
+    map_preferences: dict | GeospatialMapConfig | None = {},
```

Comment:
Same comment here about the use of mutable default values.

```suggestion
    map_preferences: dict | GeospatialMapConfig | None = None,
```